cms/middleware/toolbar.py

```diff
@@ -77,8 +77,10 @@ def process_request(self, request):
             if build in request.GET and not request.session.get('cms_build', False):
                 request.session['cms_build'] = True
         else:
```

Comment:
It's tempting to use `elif` when you see  `else` and then `if` but in this case setting `cms_edit` to `False` is not related to that if `cms_build` is `True` or not.

Both conditions need to be checked if the condition in the upper `if` fails. If I change it like:

    if not request.session.get('cms_toolbar_disabled', False) and (
                request.user.is_staff or (anonymous_on and request.user.is_anonymous())
        ):
        ...
    elif equest.session.get('cms_build'):
        request.session['cms_build'] = False
    elif if request.session.get('cms_edit'):
        request.session['cms_edit'] = False

This will have totally different behavior for `cms_edit`.

Probably I misunderstood your suggestion. What you have in mind?